lib/pg.provider.js

```diff
@@ -692,141 +474,75 @@ class PostgresProvider extends StorageProvider {
-                  if (rollbackError) {
-                    this.systemLogger(rollbackError);
-                    callback(
-                      new GSError(
-                        errorCodes.INTERNAL_PROVIDER_ERROR,
-                        rollbackError
-                      )
-                    );
-                  } else {
-                    this.systemLogger(err);
-                    callback(
-                      new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err)
-                    );
-                  }
-                  done();
-                });
-                return;
-              }
+    if (permissionChecker) {
+      await permissionChecker(category, { record: obj });
+    }
 
-              client.query('COMMIT', err => {
-                if (err) {
-                  this.systemLogger(err);
-                  callback(
-                    new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err)
-                  );
-                } else {
-                  callback(null, ctx.id);
-                }
-                done();
-              });
-            }
-          );
-        });
-      } else {
-        createRecord(category, obj, this.pool, null, (err, res) => {
-          if (err) {
-            this.systemLogger(err);
-            callback(new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err));
-          } else {
-            callback(null, res);
-          }
-        });
+    if (!isGlobalCategory(categoryDefinition)) {
+      try {
+        return await createRecord(category, obj, this.pool, null);
+      } catch (err) {
+        this.systemLogger(err);
+        throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err);
       }
-    });
+    }
+
+    let client;
+    try {
+      client = await this.pool.connect();
+    } catch (err) {
+      this.systemLogger(err);
+      throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err);
+    }
+
+    try {
+      await client.query('BEGIN');
+      const id = await this.takeId(client);
+      await Promise.all(
+        extractIncludeCategoriesData(categoryDefinition, obj).map(data =>
+          createRecord(data.category, data.value, client, id)
+        )
+      );
+      await createRecord(category, obj, client, id);
+      await client.query(
+        'UPDATE "Identifier"' +
+          ' SET "Status" = \'Actual\', "Change" = CURRENT_TIMESTAMP,' +
+          ' "Category" = (SELECT "Id" FROM "Category" WHERE "Name" = $1),' +
+          ' "Checksum" = (SELECT get_checksum($1, $2, \'sha512\'))' +
+          ' WHERE "Id" = $2',
+        [category, id]
+      );
+      return id;
+    } catch (err) {
+      this.systemLogger(err);
+      return client.query('ROLLBACK').then(
+        () => {
+          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err);
+        },
+        rollbackError => {
+          this.systemLogger(rollbackError);
+          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, rollbackError);
+        }
+      );
```

Comment:
Nit: 
Considering this pattern occurs in a few places I think we can extract it in
```js
const gsRollback = (provider, client, err) => {
  provider.systemLogger(err);
  client.query('ROLLBACK').then(
        () => {
          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err);
        },
        rollbackError => {
          provider.systemLogger(rollbackError);
          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, rollbackError);
        }
      );
}
```